pms/models/product_pricelist.py

```diff
@@ -61,27 +61,72 @@ class ProductPricelist(models.Model):
+                            ON item.id = lin.product_pricelist_item_id
+                WHERE  (lin.pms_property_id = %s OR lin.pms_property_id IS NULL)
+                   AND (cab.pms_property_id = %s OR cab.pms_property_id IS NULL)
+                   AND (item.product_tmpl_id IS NULL
+                        OR item.product_tmpl_id = ANY(%s))
+                   AND (item.product_id IS NULL OR item.product_id = ANY(%s))
+                   AND (item.categ_id IS NULL OR item.categ_id = ANY(%s))
+                   AND (item.pricelist_id = %s)
+                   AND (item.date_start IS NULL OR item.date_start <=%s)
+                   AND (item.date_end IS NULL OR item.date_end >=%s)
+                   AND (item.date_start_overnight IS NULL
+                        OR item.date_start_overnight <=%s)
+                   AND (item.date_end_overnight IS NULL
+                        OR item.date_end_overnight >=%s)
+                GROUP  BY item.id
+                ORDER  BY item.applied_on,
+                          /* REVIEW: priotrity date sale / date overnight */
+                          item.date_end - item.date_start ASC,
+                          item.date_end_overnight - item.date_start_overnight ASC,
+                          NULLIF((SELECT COUNT(1)
+                           FROM   pms_property_product_pricelist_item_rel l
+                           WHERE  item.id = l.product_pricelist_item_id)
+                          + (SELECT COUNT(1)
+                             FROM   pms_property_product_pricelist_rel c
+                             WHERE  item.pricelist_id = c.product_pricelist_id),0)
+                          NULLS LAST,
+                          item.id DESC;
+                """,
+                (
+                    self._context["property"],
+                    self._context["property"],
+                    prod_tmpl_ids,
+                    prod_ids,
+                    categ_ids,
+                    self.id,
+                    date,
+                    date,
+                    self._context["date_overnight"],
+                    self._context["date_overnight"],
+                ),
             )
-            reverse_id = items_filtered.sorted(id, reverse=True)
-            return items_filtered.sorted(
-                key=lambda s: (
-                    s.applied_on,
-                    True if (not s.date_end or not s.date_start) else False,
-                    True
-                    if (not s.date_end or not s.date_start)
-                    else (s.date_end - s.date_start).days,
-                    ((not s.pms_property_ids, s), len(s.pms_property_ids)),
-                    reverse_id,
-                )
+
+            item_ids = [x[0] for x in self.env.cr.fetchall()]
+            items = self.env["product.pricelist.item"].browse(item_ids)
+        else:
+            items = super(ProductPricelist, self)._compute_price_rule_get_items(
```

Comment:
thinking in this case ("non-overnight" rates) I think it is necessary to add (in the style as it was before) at least the prcelist items filter by properties when property comes in the context. What do you think? Otherwise it would not be possible to configure different prices per hotel in the services (e.g. breakfast - which do not have overnights)

Reply:
ok, fixed :)